libs/shared/flagd-core/src/lib/feature-flag.ts

```diff
@@ -42,14 +74,59 @@ export class FeatureFlag {
     return this._defaultVariant;
   }
 
-  get targeting(): unknown {
-    return this._targeting;
-  }
-
   get variants(): Map<string, FlagValue> {
     return this._variants;
   }
 
+  get metadata(): FlagMetadata {
+    return this._metadata;
+  }
+
+  evaluate(evalCtx: EvaluationContext): ResolutionDetailsWithFlagMetadata<JsonValue> {
+    let variant: string;
+    let reason: ResolutionReason;
+
+    if (this._targetingParseError) {
+      throw this._targetingParseError;
+    } else if (!this._targeting) {
+      variant = this._defaultVariant;
+      reason = StandardResolutionReasons.STATIC;
+    } else {
+      let targetingResolution: JsonValue;
+      try {
+        targetingResolution = this._targeting.evaluate(this._key, evalCtx);
+      } catch (e) {
+        throw new GeneralError(`Error evaluating targeting rule for flag '${this._key}'`, { cause: e });
+      }
+
+      // Return default variant if targeting resolution is null or undefined
+      if (targetingResolution === null || targetingResolution === undefined) {
```

Comment:
This is required to properly support short-circuiting json logic. If we used `!targetingResolution`, it would be true for any falsy values.